polkadot/primitives/src/v6/executor_params.rs

```diff
@@ -130,6 +182,126 @@ impl ExecutorParams {
 		}
 		None
 	}
+
+	/// Check params coherence.
+	pub fn check_consistency(&self) -> Result<(), ExecutorParamError> {
```

Comment:
Can we have this logic right in `configuration.rs`? This file seems to be mainly defining a bunch of types with not much actual logic.

Reply:
Yeah, it would be nice to if we could put the checks in one place. But that may require opening up the internal of `ExecutorParams`, do you think it's worth it? @mrcnski

Reply:
Not sure what you mean exactly, but I *think* it's fine.

Reply:
Sorry, I was thinking that we might need to make the internal state public in order to do the checks in another crate, like `pub struct ExecutorParams(pub Vec<ExecutorParam>)`. Then it's possible that other crates may rely on the internal state which is not great.

I just realized that it comes with an iterator :)

But still, for the sake of separation of concerns, I think it wouldn't be a bad idea that the type does its own checks. Like, what if the checks are needed else where (other than `configurations.rs`)?

Reply:
Fair enough. :)